src/core/hashTable.cc

```diff
@@ -515,7 +515,8 @@ void HashTable_O::sxhash_eql(HashGenerator& hg, T_sp obj) {
     return;
   }
   case gctools::single_float_tag: {
-    hg.addValue0(std::abs(::floor(obj.unsafe_single_float())));
+    float value = obj.unsafe_single_float();
+    hg.addValue0((std::fpclassify(value) == FP_ZERO) ? 0u : float_convert<float>::to_bits(value));
```

Comment:
No the issue is that standard requires that -0.0 and 0.0 behave the same in hashes.

Reply:
okay, that makes sense for the others, but for an eql hash they should be distinct, right? Since they're not `eql`? Unless there's a hash table specific rule I'm missing here.

Reply:
In addition to the tests in sxhash.lsp it is tested by gethash in all varieties.

https://gitlab.common-lisp.net/ansi-test/ansi-test/-/blob/master/hash-tables/gethash.lsp?ref_type=heads#L84-148

Reply:
Actually, I'm wrong. The only gethash test that requires that is equalp. Apologies, I'll fix it.